src/datasets/arrow_dataset.py

```diff
@@ -262,10 +262,13 @@ def from_file(
         info: Optional[DatasetInfo] = None,
         split: Optional[NamedSplit] = None,
         indices_filename: Optional[str] = None,
+        in_memory: bool = False,
     ) -> "Dataset":
         """ Instantiate a Dataset backed by an Arrow table at filename """
-        mmap = pa.memory_map(filename)
-        f = pa.ipc.open_stream(mmap)
+        # Stream backed by memory-mapped file / file descriptor
+        stream_from = pa.memory_map if not in_memory else pa.input_stream
+        stream = stream_from(filename)
+        f = pa.ipc.open_stream(stream)
         pa_table = f.read_all()
         data_files = [{"filename": filename}]
```

Comment:
```suggestion
        data_files = [{"filename": filename}] if not in_memory else None
```
We differentiate in-memory datasets from on-disk datasets by looking at `data_files`. If it's empty, then the dataset is in-memory.

We might change that at some point though since it's not flexible in my opinion.